be/src/exprs/vectorized/cast_expr.cpp

```diff
@@ -1052,10 +1196,49 @@ Expr* VectorizedCastExprFactory::from_thrift(const TExprNode& node) {
             CASE_TO_STRING_FROM(TYPE_DECIMAL32);
             CASE_TO_STRING_FROM(TYPE_DECIMAL64);
             CASE_TO_STRING_FROM(TYPE_DECIMAL128);
+            CASE_TO_STRING_FROM(TYPE_JSON);
         default:
             LOG(WARNING) << "vectorized engine not support from type: " << from_type << ", to type: " << to_type;
             return nullptr;
         }
+    } else if (from_type == TYPE_JSON || to_type == TYPE_JSON) {
+        // TODO(mofei) simplify type enumeration
+        if (from_type == TYPE_JSON) {
+            switch (to_type) {
+                CASE_FROM_JSON_TO(TYPE_BOOLEAN);
+                CASE_FROM_JSON_TO(TYPE_TINYINT);
+                CASE_FROM_JSON_TO(TYPE_SMALLINT);
+                CASE_FROM_JSON_TO(TYPE_INT);
+                CASE_FROM_JSON_TO(TYPE_BIGINT);
+                CASE_FROM_JSON_TO(TYPE_LARGEINT);
+                CASE_FROM_JSON_TO(TYPE_FLOAT);
+                CASE_FROM_JSON_TO(TYPE_DOUBLE);
+                CASE_FROM_JSON_TO(TYPE_JSON);
+            default:
+                LOG(WARNING) << "vectorized engine not support from type: " << from_type << ", to type: " << to_type;
```

Comment:
we could add a DCHECK here

Reply:
I suppose not, FE does not check the type casting, so any `cast(xxx as yyy)` could happen here, the illegal cast should not made the BE crash.

Reply:
other expressions assume that the return of a child expression must be non-null, so this branch must generate a segment fault. So I suggest adding a DCHECK

Reply:
```
ColumnPtr ExprContext::evaluate(Expr* e, vectorized::Chunk* chunk) {
#ifndef NDEBUG
    if (chunk != nullptr) {
        chunk->check_or_die();
        CHECK(!chunk->is_empty());
    }
#endif
    auto ptr = e->evaluate(this, chunk);
    DCHECK(ptr != nullptr);
    if (chunk != nullptr && 0 != chunk->num_columns() && ptr->is_constant()) {
        ptr->resize(chunk->num_rows());
    }
    return ptr;
}
```
I think it would be a more friendly idea to have the error exposed earlier.

Reply:
It's not the `Expr::evaluate` path, but the `Expr::create_vectorized_expr`, which could return nullptr.

The `Expr::create_vectorized_expr` checks whether the expr is nullptr, transform it into an error if true.

The call stack is:
```c++
Expr::create_vectorized_expr() {
  expr = VectorizedCastExprFactory::from_thrift()
  if (*expr == nullptr) {
    LOG(WARNING) << "Vectorized engine node type return nullptr: " + std::to_string(texpr_node.node_type);
    return Status::InternalError("Vectorized engine does not support the operator");
  }
}
```